docs/docker_run.rst

```diff
@@ -0,0 +1,190 @@
+
+A ``TemporaryVolume`` class is provided representing a temporary directory on the host
+machine that is mounted into the container. ``TemporaryVolume.default`` holds a default
+instance that is used as the default location for many other parts of the girder_worker docker
+infrastructure for example when downloading file. See `Downloading files from Girder`_.
+However, it can also be using explicity for example, here it is being passed as a
+container argument for use within a container. Again, girder_worker will take care
+of transforming the ``TemporaryVolume`` instance into the appropriate path inside
+the container, so the container entrypoint will simply recieve a path.
+
+.. code-block:: python
+
+    vol = Volume('/home/docker/data', '/mnt/docker/')
+    docker_run.delay('my/image', pull_image=True, container_args=[TemporaryVolume.default])
+
+Note that we don't have to add the instance to the ``volumes`` parameter as it is
+automatically added to the list of volumes to mount.
+
+Downloading files from Girder
+-----------------------------
+
+Accessing files held in girder from within a container is straight forward using
+the ``GirderFileIdToVolume`` utility class. One simply provides the file id as
+a argument to the constructor and pass the instance as a container argument.
+
+.. code-block:: python
+
+    docker_run.delay('my/image', pull_image=True,
+        container_args=[GirderFileIdToVolume(file_id)])
+
+The ``GirderFileIdToVolume`` instance will take care of downloading the file from
+Girder and passes the path it was download to the docker containers entrypoint as an argument.
+
+If no ``volume`` parameter is specified then the file will be downloading to the
+task temporary volume. The file can also be downloaded to a specific ``Volume`` by
+specifying a volume parameter, as follows:
+
+.. code-block:: python
+
+    vol = Volume(host_path, container_path)
+    docker_run.delay('my/image', pull_image=True,
+        container_args=[GirderFileIdToVolume(file_id,volume=vol)])
+
+If the being downloaded is particularly large you may want to consider streaming
+it into the container using a named pipe. See `Streaming Girder files into a container`_
+for more details.
+
+Uploading files to Girder items
+-------------------------------
+
+Utility classes are also provide to simplify uploading files generated by a the
+docker container. The ``GirderUploadVolumePathToItem`` provides the functionality
+to upload a file to an item. In the example below, we using the ``VolumePath``
+utility class to define a file path that we then pass to the docker container.
+The docker container can write data to this file path. As well as passing the
+``VolumePath`` instance as a container argument we also pass it to ``GirderUploadVolumePathToItem``
+, the ``GirderUploadVolumePathToItem`` instance is add to ``girder_result_hooks``. This
+tell girder_work to upload the file path to the item id provided when the docker
+container has finished running.
```

Comment:
I am having trouble parsing the last two sentences of this paragraph, maybe needs to be reworded